app/views/memberships/_applications_with_roles.html.erb

```diff
@@ -0,0 +1,36 @@
+<div class="field">
+  <h3>Roles for Applications</h3>
+
+  <% if @organisation.applications.empty? %>
+    <p><%= t("organisation_has_no_applications", organisation_name: @organisation.name) %></p>
+  <%- end %>
+
+  <dl>
+    <% @organisation.applications.each_with_index do |application, i| %>
+    <dt><%= application.name %></dt>
+    <dd>
+      <%= hidden_field_tag "membership[applications_with_roles[#{i}[name]]]", application.name %>
+
+      <% if application.handles_own_authorization %>
+
+        <%= label_tag do %>
+          <%= check_box_tag "membership[applications_with_roles[#{i}[can_login]]", true, membership.applications_with_roles.any? {|app| app["name"] == application.name && app["can_login"]} %>
```

Comment:
Should these actually be nested? I thought it would be something like `membership[applications_with_roiles][#{i}][can_login]`?